sklearn/ensemble/_stacking.py

```diff
@@ -182,21 +189,8 @@ def fit(self, X, y, sample_weight=None):
         ]
 
         X_meta = self._concatenate_predictions(X, predictions)
-        if sample_weight is not None:
-            try:
-                self.final_estimator_.fit(
-                    X_meta, y, sample_weight=sample_weight
-                )
-            except TypeError as exc:
-                if "unexpected keyword argument 'sample_weight'" in str(exc):
-                    raise TypeError(
-                        "Underlying estimator {} does not support sample "
-                        "weights."
-                        .format(self.final_estimator_.__class__.__name__)
-                    ) from exc
-                raise
-        else:
-            self.final_estimator_.fit(X_meta, y)
+        _parallel_fit_estimator(self.final_estimator_, X_meta, y,
```

Comment:
I would put a comment because `_parallel` is a weird name since we fit a single estimator but this is worth.
However, I would propose a subsequent maintenance PR to rename `_parallel_fit_estimator` to `_fit_single_estimator`.
The parallel part will come when we call `Parallel` on the `fit_single_estimator`

Reply:
Good idea, I thought this naming was a little strange as well.

I added a TODO comment to the docstring of `_parallel_fit_estimator` and linked to this review comment. It seems like a simple enough change that I may tackle it as a maintenance fix in a follow-up PR.

Reply:
Created a follow-up issue for reference in the maintenance PR that will do the renaming: https://github.com/scikit-learn/scikit-learn/issues/16595

Reply:
Change made in this PR: https://github.com/scikit-learn/scikit-learn/pull/16599